src/components/Text/Text.js

```diff
@@ -8,17 +8,29 @@ import { sizes } from '../../styles/constants';
 
 const { KILO, MEGA, GIGA } = sizes;
 
+const mobileSizeMap = {
+  [KILO]: KILO,
+  [MEGA]: MEGA,
+  [GIGA]: MEGA
+};
+
 const baseStyles = ({ theme }) => css`
   label: text;
   font-weight: ${theme.fontWeight.regular};
   margin-bottom: ${theme.spacings.mega};
 `;
 
-const sizeStyles = ({ theme, size }) => css`
-  label: text--${size};
-  font-size: ${theme.typography.text[size].fontSize};
-  line-height: ${theme.typography.text[size].lineHeight};
-`;
+const sizeStyles = ({ theme, size }) =>
+  css`
+    label: text--${size};
+    font-size: ${theme.typography.text[mobileSizeMap[size]].fontSize};
```

Comment:
The above suggestion could be further abstracted by passing in the size map as an argument so you can use it for all components requiring different size mappings on mobile.